core/libp2p/host/basic_host/basic_host.cpp

```diff
@@ -31,7 +31,7 @@ namespace libp2p::host {
   }
 
   peer::PeerInfo BasicHost::getPeerInfo() const {
-    return {getId(), getObservedAddresses()};
+    return {getId(), getAddresses()};
```

Comment:
maybe concat all addresses (getObservedAddrs, getAddrs, getAddrsInterfaces...) into a set and use a set here?

Reply:
Combined unique addresses from all 3 methods.